src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadStacktrace.tsx

```diff
@@ -1,24 +1,48 @@
-import {ExceptionValue} from 'app/types';
+import {ExceptionType} from 'app/types';
 import {Event} from 'app/types/event';
 import {Thread} from 'app/types/events';
 
 import getThreadException from './getThreadException';
 
-function getThreadStacktrace(thread: Thread, event: Event, raw: boolean) {
-  const exc = getThreadException(thread, event);
-  if (exc) {
-    let rv: ExceptionValue['stacktrace'] | undefined = undefined;
-
-    for (const singleExc of exc.values) {
-      if (singleExc.threadId === thread.id) {
-        rv = singleExc.stacktrace;
-        if (raw && singleExc.rawStacktrace) {
-          rv = singleExc.rawStacktrace;
-        }
-      }
+function getThreadStacktrace(
+  event: Event,
+  raw: boolean,
+  thread?: Thread,
+  exception?: Required<ExceptionType>,
+  checkException?: boolean
+) {
```

Comment:
is it possible to add an explicit return type, seems like there's a few options

Reply:
the only options are `undefined` and `StacktraceType`, but I can add an explicit return type 😉